src/bin/cargo/main.rs

```diff
@@ -28,6 +29,13 @@ fn main() {
         }
     };
 
+    let nightly_features_allowed = matches!(&*features::channel(), "nightly" | "dev");
+    if nightly_features_allowed {
+        clap_complete::CompleteEnv::with_factory(|| cli::cli(&mut gctx))
```

Comment:
This is using the built-in shell support, expanding our shell coverage from bash and zsh to also include elvish, fish, and powershell.

Do we want to do that?  Even if we experiment with this, we would need to track that in the PR description and the tracking issue.

Reply:
> By using clap_complete crate, we could extend the supported shells to Bash, Zsh, Elvish, Fish, and PowerShell. However, at the current stage, the support for PowerShell in clap_complete is not fully developed, so this PR does not include any content related to PowerShell.

Thanks for adding this to the PR. I've added a similar Open Questioln to #14520.

Note: currently, you are supporting `powershell`, you just might not be documenting it.

Reply:
Also, I'm fine with us doing powershell for now, we just need to evaluate whether to stabilize it later.

Exposing it now gets more visibility for more testing and feedback rather than waiting for some point for it to be ready enough.

Reply:
If we support `powershell` for now. I need to add test for `powershell` and first add PowerShell support to the completest crate (continuing my previous work).

Reply:
Having it validated *somehow* is a blocker for stabilization.  Note that none of our existing completions are validated in an automated way, so automating it is not necessarily a requirement.

Reply:
I added instructions on how to use the feature in PowerShell to `unstable.md`. Call for everybody to try this new feature.

As for testing, I haven't added tests for PowerShell yet. I will add tests once PowerShell support in the `clap_complete` crate stabilizes.